### PR TITLE
[compiler-rt][rtsan] Introduce RTSAN_OPTIONS and flags

### PR DESCRIPTION
**Pull request by cjappl**
All of this matches boilerplate you can find in the other sanitizers. 

This allows us to specify some of the existing flags like:
* abort_on_error - get better stack traces on mac, speed up our unit tests
* verbosity - help debugging
* color - colorized output reports

At some point we will want custom flags, such as:
* halt_on_error - continue mode, matching other sanitizers name for this
* print_exit_stats - how many rt violations occurred during continue mode?? 
* ?? what else?? 

This lays down the basics necessary to do these in the future

### CODE REVIEW

**Comment by cjappl** [comment]
CC @davidtrevelyan for review

File: compiler-rt/lib/rtsan/rtsan_stack.cpp [hunk 1]
```diff
@@ -29,21 +29,9 @@ void BufferedStackTrace::UnwindImpl(uptr pc, uptr bp, void *context,
 }
 } // namespace __sanitizer
 
-static void SetGlobalStackTraceFormat() {
```
Comment (cjappl) [inline compiler-rt/lib/rtsan/rtsan_stack.cpp:32]: This was moved to InitializeFlags

File: compiler-rt/lib/rtsan/rtsan_flags.cpp [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+static void RegisterRtsanFlags(FlagParser *parser, Flags *f) {
+#define RTSAN_FLAG(Type, Name, DefaultValue, Description)                      \
+  RegisterFlag(parser, #Name, Description, &f->Name);
+#include "rtsan_flags.inc"
+#undef RTSAN_FLAG
+}
+
+void Flags::SetDefaults() {
```
Comment (davidtrevelyan) [inline compiler-rt/lib/rtsan/rtsan_flags.cpp:33]: Can we eliminate this function if we change line 17 of rtsan_flags.h to the following?

```cpp
#define RTSAN_FLAG(Type, Name, DefaultValue, Description) Type Name { DefaultValue };
```

Comment (cjappl) [reply compiler-rt/lib/rtsan/rtsan_flags.cpp:33]: Wow, that is... extremely clever!! I had to think through what these macros are doing to figure out what was going on here.

This does work.

One possible downside of this approach, is the other sanitizers all seem to agree on the other approach, e.g.

```
> rg "define.*SAN_FLAG\(Type" -B 2
compiler-rt/lib/nsan/nsan_flags.h
16-
17-struct Flags {
18:#define NSAN_FLAG(Type, Name, DefaultValue, Description) Type Name;

compiler-rt/lib/nsan/nsan_flags.cpp
26-
27-void Flags::SetDefaults() {
28:#define NSAN_FLAG(Type, Name, DefaultValue, Description) Name = DefaultValue;

...

compiler-rt/lib/asan/asan_flags.h
30-
31-struct Flags {
32:#define ASAN_FLAG(Type, Name, DefaultValue, Description) Type Name;

compiler-rt/lib/asan/asan_flags.cpp
36-
37-void Flags::SetDefaults() {
38:#define ASAN_FLAG(Type, Name, DefaultValue, Description) Name = DefaultValue;

compiler-rt/lib/ubsan/ubsan_flags.h
22-
23-struct Flags {
24:#define UBSAN_FLAG(Type, Name, DefaultValue, Description) Type Name;

compiler-rt/lib/ubsan/ubsan_flags.cpp
34-
35-void Flags::SetDefaults() {
36:#define UBSAN_FLAG(Type, Name, DefaultValue, Description) Name = DefaultValue;
```

Whether this is by design or not, I'm not exactly sure. I want to do a second round of reviews with a broader audience on this one, possibly post-commit where we could ask a question like this.

Comment (cjappl) [reply compiler-rt/lib/rtsan/rtsan_flags.cpp]: I have updated this to your proposed version in the latest.

**Pull request merged**